src/models/goods-vehicle-movements.js

```diff
@@ -35,19 +31,33 @@ const mapGmrDeclaration = (customsDeclarations, gvmDeclaration) => {
   }
 }
 
+const incrementCounter = (counter, gmrDeclaration, shouldBeCounted) => {
+  if (gmrDeclaration.isKnownMrn === shouldBeCounted) {
+    ++counter
+  }
+
+  return counter
+}
+
 const mapCustomsDeclarations = (
   customsDeclarations,
   goodsVehicleMovement
 ) => {
   const gmrCustoms = goodsVehicleMovement?.declarations?.customs?.map((custom) => {
     return mapGmrDeclaration(customsDeclarations, custom)
-  })
+  }) || []
 
   const gmrTransits = goodsVehicleMovement?.declarations?.transits?.map((transit) => {
     return mapGmrDeclaration(customsDeclarations, transit)
-  })
+  }) || []
 
-  return (gmrCustoms || []).concat(gmrTransits || [])
+  return {
+    mrnCounts: {
+      knownMrns: gmrCustoms.reduce((knownMrnsCount, custom) => incrementCounter(knownMrnsCount, custom, true), 0),
```

Comment:
if you feel it makes the code more legible/concise then `knownMrns` and `unknownMrns` can be renamed to `known` and `unknown` given that the containing object is entitled `mrnCounts`.